src/main/java/io/confluent/kafkarest/SimpleConsumerFactory.java

```diff
@@ -59,16 +58,40 @@ private String nextClientId() {
     return id.toString();
   }
 
-  public SimpleConsumer createConsumer(final String host, final int port) {
+  public ConsumerProvider createConsumer() {
 
     final String clientId = nextClientId();
 
-    log.debug("Creating SimpleConsumer with id " + clientId + " (host: " + host + ", port: " + port + ")");
-    return new SimpleConsumer(
-        host, port,
-        simpleConsumerConfig.socketTimeoutMs(),
-        simpleConsumerConfig.socketReceiveBufferBytes(),
-        clientId);
+    log.debug("Creating SimpleConsumer with id " + clientId);
+    Properties properties = (Properties) simpleConsumerProperties.clone();
+
+    properties.setProperty("client.id", clientId);
+    properties.setProperty("max.poll.records", String.valueOf(
```

Comment:
Sorry for the long comment, but this is probably the main work to sort out in this patch.

There is some subtlety with the interplay of max.poll.records and the cache that you've included below. Although you request only `max.poll.records` from the consumer, it will fetch an amount which is bound by `max.partition.fetch.bytes`, which may contain considerably more records than what was asked for. It will then feed out this data in batches of size max.poll.records when poll() is called.

This makes the configuration `cache.max.records` a little misleading since it doesn't account for the data cached in the consumer. In fact, there is an advantage to leaving the data in the consumer as long as possible. If the fetched data is compressed, then we will not decompress it until it is ready to be returned to the user, which helps to keep memory usage lower.

The other interesting thing is the impact of prefetching. Once the consumer's cached records are depleted, it will send a new round of fetches prior to returning from a poll() call. If we end up seeking to another offset before polling again, those fetches will have to be discarded. In the worst case, with multiple users fetching the same topic partition with the same consumer instance, we'll have to discard fetch requests from every call to poll(). So it might be nice to have more stickiness between the user and the KafkaConsumer instance. We could potentially do this by remembering the position of the consumer in the cache. Then instead of mapping to a consumer instance only by the topic partition, we can take into account the requested offset. For each topic partition, then we would have multiple consumers and we'd only need to do a seek when we've reached some limit.

Thoughts?